app/models/concerns/waste_carriers_engine/can_change_registration_status.rb

```diff
@@ -77,13 +77,17 @@ def log_status_change
     def renewal_allowed?
       return false unless %w[ACTIVE EXPIRED].include?(status)
       return true if renewal_application_submitted?
+      return true if renew_expired_registration?
+
+      close_to_expiry_date? && should_not_be_expired?
+    end
+
+    def renew_expired_registration?
       return true if in_expiry_grace_window?
 
       # The only time an expired registration can be renewed is if the application has previously been submitted,
       # or it is withion the grace window - otherwise expiry is an automatic no
       return false if EXPIRED?
```

Comment:
Now that this isn't followed by further checks, there's nothing to return `true` if it's not `EXPIRED?` – I think we would get `nil` instead. Which makes me wonder if the tests are covering this.